### PR TITLE
Neovim darwin

### PR DESCRIPTION
**Pull request by LnL7**
neovim and some of it's dependencies where broken on darwin.

### CODE REVIEW

File: pkgs/top-level/lua-packages.nix [hunk 1]
```diff
@@ -33,8 +33,15 @@ let
       sha256 = "16fffbrgfcw40kskh2bn9q7m3gajffwd2f35rafynlnd7llwj1qj";
     };
 
+    buildFlags = stdenv.lib.optionalString stdenv.isDarwin "macosx";
+
+    patchPhase = stdenv.lib.optionalString stdenv.isDarwin ''
+      sed -e "s,MACOSX_DEPLOYMENT_TARGET=10.4,MACOSX_DEPLOYMENT_TARGET=10.10," -i Makefile
```
Comment (spwhitt) [inline pkgs/top-level/lua-packages.nix]: Did you try without this line? It may not be necessary.

File: pkgs/top-level/lua-packages.nix [hunk 2]
```diff
@@ -33,8 +33,11 @@ let
       sha256 = "16fffbrgfcw40kskh2bn9q7m3gajffwd2f35rafynlnd7llwj1qj";
     };
 
+    buildFlags = stdenv.lib.optionalString stdenv.isDarwin "macosx";
+
     preBuild = ''
       makeFlagsArray=(
+        ${stdenv.lib.optionalString stdenv.cc.isClang "CC=clang"}
```
Comment (gridaphobe) [inline pkgs/top-level/lua-packages.nix]: Might as well make this an unconditional `CC=$CC` as we define `CC` in the stdenv.

**Comment by LnL7** [comment]
I rebased, now it includes python support from master

**Comment by garbas** [comment]
merged [here](http://git.io/vIysd)

**Pull request closed without merging**